zipline/pipeline/loaders/utils.py

```diff
@@ -260,3 +275,39 @@ def check_data_query_args(data_query_time, data_query_tz):
                 data_query_tz,
             ),
         )
+
+
+def zip_with_floats(dates, flts):
+        return pd.Series(flts, index=dates).astype('float')
+
+
+def num_days_between(dates, start_date, end_date):
+    return num_days_in_range(dates, start_date, end_date)
+
+
+def zip_with_dates(index_dates, dts):
+    return pd.Series(pd.to_datetime(dts), index=index_dates)
+
+
+def get_values_for_date_ranges(zip_date_index_with_vals,
+                               vals_for_date_intervals,
+                               date_intervals,
+                               date_index):
+    """
+    Returns a
+    :param zip_vals_for_date_index:
+    :param num_days_between_dates:
+    :param vals_for_date_intervals:
+    :param date_intervals:
+    :return:
+    """
+    # Fill in given values for given date ranges.
+    return zip_date_index_with_vals(
+        date_index,
+        list(
+            itertools.chain(*[
```

Comment:
@mtydykov what I mean is that this is taking a bunch of numpy-ish datastructures (meaning, either numpy or pandas objects) and producing a numpy-ish datastructure, but all of the data processing here is being done on sequences of python objects.  That generally means we're losing most or all of the performance benefits of working with numpy objects in the first place.